frontend/pulse3d/src/pages/uploads.js

```diff
@@ -272,37 +277,35 @@ export default function Uploads() {
 
   useEffect(() => {
     getAllJobs();
-    // start 10 second interval
-    const uploadsInterval = setInterval(() => getAllJobs(), [1e4]);
     // don't call get jobs if downloading or deleting in progress because it backs up server
-    if (!["downloading", "deleting"].includes(modalState)) {
-      toggleUpdateData(!updateData);
+    if (!["downloading", "deleting"].includes(modalState) && uploads.length > 0) {
+      if (!statusUpdateInterval) {
+        statusUpdateInterval = setInterval(async () => {
+          if (!["downloading", "deleting"].includes(modalState)) {
+            await getAllJobs();
+          }
+        }, [1e4]);
+      }
     }
-
-    //clear interval when switching pages
-    return () => clearInterval(uploadsInterval);
```

Comment:
why don't you add this back in instead of passing clearInterval to the layout? the return clears the interval when you switch pages which I think is what you're doing by adding it to the onClick of the accordion tabs